Update Android Gradle Plugin version to 8.11.1

mikescamell opened the pull request:
Task/Issue URL: https://app.asana.com/1/137249556945/project/1202552961248957/task/1210802695070486

### Description

This PR updates the Gradle wrapper from 8.10.2 to 8.14.3 and the Android Gradle Plugin from 8.7.3 to 8.11.1. The changes include:

### Steps to test this PR

_Verify build functionality_
- [ ] Run `./gradlew clean build` to ensure the new Gradle version works correctly
- [ ] Verify that Android Studio recognizes the updated Gradle version
- [ ] Check that all CI builds pass with the updated configuration

mikescamell committed:
Update Android Gradle Plugin version to 8.11.1

mikescamell committed:
Update Gradle wrapper to version 8.14.3

mikescamell committed:
Remove Kotlinter

mikescamell committed:
Update spotless to latest version

mikescamell committed:
Add Kotlin formatting to Spotless

ratchetFrom means Spotless will apply only to files which have changed since origin/main

It also means we need to make some changes to CI due to ensure builds pass due to shallow clones. See https://github.com/diffplug/spotless/tree/main/plugin-gradle#using-ratchetfrom-on-ci-systems for more details.

mikescamell committed:
Update documentation

mikescamell committed:
Remove deprecated indentWithSpaces

mikescamell committed:
Fix ignored files committed

Spotless breaks because we have classes and files that have been commited that should technically be ignored by our .gitignore settings. See https://github.com/diffplug/spotless/issues/911#issuecomment-904887074

Therefore we have to either remove these files or move them.

In this case the DeviceShield stuff was in a package called "report" which we ignore in .gitignore. Therefore I gave it a new package name

We want to keep the icon and I guess we want to keep the copyright folder so I made that work in .gitignore

mikescamell committed:
Format the classes edited due to moving package changes

As these have "changed" since develop then they need formatting!

mikescamell committed:
Update Spotless plugin version to 7.2.0

mikescamell committed:
WIP

daxmobile commented on .github/workflows/ci.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>label "android-large-runner" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2022", "windows-2019", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "ubuntu-20.04", "macos-latest", "macos-latest-xl", "macos-latest-xlarge", "macos-latest-large", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xl", "macos-14-xlarge", "macos-14-large", "macos-14", "macos-13-xl", "macos-13-xlarge", "macos-13-large", "macos-13", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file [runner-label]

https://github.com/duckduckgo/Android/blob/4b21042cf92d8f300733b7d0ebc36e2e9e1e738a/.github/workflows/ci.yml#L47
<!-- __reviewdog__:ChA3MjBjOTI2ZTE4OGMzYzE4EgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/ci.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>label "android-large-runner" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2022", "windows-2019", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "ubuntu-20.04", "macos-latest", "macos-latest-xl", "macos-latest-xlarge", "macos-latest-large", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xl", "macos-14-xlarge", "macos-14-large", "macos-14", "macos-13-xl", "macos-13-xlarge", "macos-13-large", "macos-13", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file [runner-label]

https://github.com/duckduckgo/Android/blob/4b21042cf92d8f300733b7d0ebc36e2e9e1e738a/.github/workflows/ci.yml#L80
<!-- __reviewdog__:ChBiMjM1OGQyMmI4MTEzMTAwEgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/ci.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>shellcheck reported issue in this script: SC2002:style:1:5: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead [shellcheck]

https://github.com/duckduckgo/Android/blob/4b21042cf92d8f300733b7d0ebc36e2e9e1e738a/.github/workflows/ci.yml#L189
<!-- __reviewdog__:ChAyZDNhMTQ0YzFmNDExNDdhEgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/ci.yml:
📝 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>shellcheck reported issue in this script: SC2086:info:1:6: Double quote to prevent globbing and word splitting [shellcheck]

https://github.com/duckduckgo/Android/blob/5880601504b9c3c3511689c76fd8218ecbc43502/.github/workflows/ci.yml#L145
<!-- __reviewdog__:ChA0MzA0OTk2OGVkMDg2ZTA1EgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/nightly.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>shellcheck reported issue in this script: SC2002:style:1:5: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead [shellcheck]

https://github.com/duckduckgo/Android/blob/5880601504b9c3c3511689c76fd8218ecbc43502/.github/workflows/nightly.yml#L153
<!-- __reviewdog__:ChAzOWFlYjQ3MGFhY2M5Yjc4EgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/ci.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>label "android-large-runner" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2022", "windows-2019", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "ubuntu-20.04", "macos-latest", "macos-latest-xl", "macos-latest-xlarge", "macos-latest-large", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xl", "macos-14-xlarge", "macos-14-large", "macos-14", "macos-13-xl", "macos-13-xlarge", "macos-13-large", "macos-13", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file [runner-label]

https://github.com/duckduckgo/Android/blob/3959791a5f0646da76bafacac0970be7c4952569/.github/workflows/ci.yml#L80
<!-- __reviewdog__:ChBiMjM1OGQyMmI4MTEzMTAwEgphY3Rpb25saW50 -->

daxmobile commented on .github/workflows/ci.yml:
🚫 **[actionlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>label "android-large-runner" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2022", "windows-2019", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "ubuntu-20.04", "macos-latest", "macos-latest-xl", "macos-latest-xlarge", "macos-latest-large", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xl", "macos-14-xlarge", "macos-14-large", "macos-14", "macos-13-xl", "macos-13-xlarge", "macos-13-large", "macos-13", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file [runner-label]

https://github.com/duckduckgo/Android/blob/3959791a5f0646da76bafacac0970be7c4952569/.github/workflows/ci.yml#L117
<!-- __reviewdog__:ChAyYzNjMGQwNmU1MmZkMDJiEgphY3Rpb25saW50 -->

mikescamell committed:
Fix broken Room code generation

We use Kotlin CodeGen for Room in the app module, but not in other modules. This along with KSP2 breaks the update to Kotlin 2.2.0

We can turn KSP2 off for now and ensure that any module that adds the Room compiler generates Kotlin code.

This will be the default in Kotlin 2.7.0+ but we cannot use it right now as we get occasional build issues. See the Asana task for more info.

mikescamell committed:
Refactor build.gradle to use plugins block for Kotlin and Java library

I could remove this long ago when changing to plugin blocks, now it works

mikescamell committed:
Remove unused Kotlin stdlib dependency from build.gradle

This gets added by default, not need for us to specify it and avoids any issues.

mikescamell committed:
Update Kotlin to 2.2.0 & KSP to 2.2.0-2.0.2

Removed the extra version.kotlin declared as I believe it's redundant

mikescamell committed:
Update Anvil to 2.6.1, Dagger to 2.56.2

This also required some code changes to remove some deprecated classes.
